### DISCUSSION TITLE
Template Form Message

### DISCUSSION THREAD

**Discussion by smartos99** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#64] [Q&A] [answered]
Hi, 
I have create it a simple form template message with two checkmarks, and sending it through SendTextMessageTemplate function. And I'm getting the following error. "Please specify a flow button component within template['components']".

Alson, this happens when I run the following code in Postman:
'{
  "messaging_product": "whatsapp",
  "to": "{{Revipient-Phone-Number}}",
  "type": "template",
  "template": {
    "name": "{{Template-Name}}",
    "language": {
      "code": "en_US"
    },
  }
}';

Error:
{
    "error": {
        "message": "(#131008) Required parameter is missing",
        "type": "OAuthException",
        "code": 131008,
        "error_data": {
            "messaging_product": "whatsapp",
            "details": "Please specify a flow button component within template['components']"
        },
        "fbtrace_id": "AhBvdjX5XQ5SPu9TxhDKXzY"
    }
}

What am I missing here? 

Thank you in advance.

**Comment by gabrieldwight** [comment, marked as answer]
Hi, I guess you are trying to send a flow template message. You can try to use this function **SendFlowMessageTemplateAsync**

```C#
FlowTemplateMessageRequest flowTemplateMessageRequest = new FlowTemplateMessageRequest();
flowTemplateMessageRequest.To = "phonenumber";
flowTemplateMessageRequest.Template = new();
flowTemplateMessageRequest.Template.Name = "template name";
flowTemplateMessageRequest.Template.Language = new();
flowTemplateMessageRequest.Template.Language.Code = LanguageCode.English_US;
flowTemplateMessageRequest.Template.Components = new List<FlowMessageComponent>()
{
    new FlowMessageComponent()
    {
        Type = "button",
        SubType = "flow",
        Index = 0,
        Parameters = new List<FlowTemplateMessageParameter>()
        {
            new FlowTemplateMessageParameter()
            {
                Type = "action",
                Action = new FlowTemplateMessageAction()
                {
                    FlowToken = "",
                }
            }
        }
    }
};

var results = await _whatsAppBusinessClient.SendFlowMessageTemplateAsync(flowTemplateMessageRequest);
```

**Reply by smartos99** [reply]
That worked. Thank you!
